src/vk/vk-posts/vkpostsyncadaptor.cpp

```diff
@@ -0,0 +1,472 @@
+        incrementSemaphore(accountId);
+        setupReplyTimeout(accountId, reply);
+    } else {
+        SOCIALD_LOG_ERROR("error: unable to request home posts from VK account with id:" << accountId);
+    }
+}
+
+void VKPostSyncAdaptor::finishedPostsHandler()
+{
+    QNetworkReply *reply = qobject_cast<QNetworkReply*>(sender());
+    bool isError = reply->property("isError").toBool();
+    int accountId = reply->property("accountId").toInt();
+
+    QByteArray replyData = reply->readAll();
+    disconnect(reply);
+    reply->deleteLater();
+    removeReplyTimeout(accountId, reply);
+
+    bool ok = false;
+    QJsonObject parsed = parseJsonObjectReplyData(replyData, &ok);
+
+    if (!isError && ok && parsed.contains(QStringLiteral("response"))) {
+        QJsonObject responseObj = parsed.value(QStringLiteral("response")).toObject();
+        QJsonArray items = responseObj.value(QStringLiteral("items")).toArray();
+        if (!items.size()) {
+            SOCIALD_LOG_DEBUG("no feed posts received for account:" << accountId);
+            decrementSemaphore(accountId);
+            return;
+        }
+
+        QJsonArray profileValues = responseObj.value(QStringLiteral("profiles")).toArray();
+        QList<UserProfile> userProfiles;
+        foreach (const QJsonValue &entry, profileValues) {
+            UserProfile profile = UserProfile::fromJsonObject(entry.toObject());
+            userProfiles << profile;
+        }
+
+        QJsonArray groupValues = responseObj.value(QStringLiteral("groups")).toArray();
+        QList<GroupProfile> groupProfiles;
+        foreach (const QJsonValue &entry, groupValues) {
+            GroupProfile profile = GroupProfile::fromJsonObject(entry.toObject());
+            groupProfiles << profile;
+        }
+
+        foreach (const QJsonValue &entry, items) {
+            QJsonObject object = entry.toObject();
+            if (!object.isEmpty()) {
+                if (object.value(QStringLiteral("type")).toString() == QStringLiteral("post")) {
+                    m_postsToAdd.append(PostData(accountId, object, userProfiles, groupProfiles));
+                } else  if (object.value(QStringLiteral("type")).toString() == QStringLiteral("photo")) {
+                    m_photoPostsToAdd.append(PostData(accountId, object, userProfiles, groupProfiles));
+                } else if (object.value(QStringLiteral("type")).toString() == QStringLiteral("photo_tag")) {
+                    SOCIALD_LOG_DEBUG("TODO: unhandled newsfeed item type:" << object.value(QStringLiteral("type")).toString() << ", skipping.");
+                } else if (object.value(QStringLiteral("type")).toString() == QStringLiteral("wall_photo")) {
+                    SOCIALD_LOG_DEBUG("TODO: unhandled newsfeed item type:" << object.value(QStringLiteral("type")).toString() << ", skipping.");
+                } else if (object.value(QStringLiteral("type")).toString() == QStringLiteral("note")) {
```

Comment:
we should probably handle note and wall_photo post types if possible.  It shouldn't be too tricky, I hope?